backend/src/main/java/planit/people/preparation/Services/Service_Calendar.java

```diff
@@ -185,17 +219,93 @@ private Map<Entity_EventPreset, List<Entity_PresetAvailability>> getAvailabiliti
-                    )
-            );
+            availabilityList.add(new Entity_PresetAvailability(availability.getId_preset_availability(), availability.getDay(), availability.getDay_off(), availability.getStart_available_time(), availability.getEnd_available_time()));
             result.put(presetKey, availabilityList);
         }
         return result;
     }
+
+    /**
+     * Delete Preset Detail. the deletion process can happen in 2 ways
+     * 1 - Delete all instances of a preset. this process is used for either
+     *      a- the preset detail update process,
+     *      b- there is only one owner of the preset,
+     *      c- there is no owner of the preset.
+     * 1.1 - In case of a and b,we first delete all PlanIt User instances in order to not violate the constraints in the PlanItUser-EventPreset helper table.
+     * 1.2 - Delete the PlanItEvent record which would trigger the deletion of all related Guest and Availability records
+     * 2 - Delete only the instance for the given PlanItUserId
+     *
+     * @param planItUserId the id of the PlanIt user whose instance should be removed.
+     * @param presetId the id of the EventPreset which should be removed.
+     */
+    public void deletePreset(Long planItUserId, Long presetId) {
+        Entity_EventPreset presetToBeDeleted = idaoEventPreset.findById(presetId).orElse(null);
+        System.out.println("presetToBeDeleted: " + presetToBeDeleted);
+        if (presetToBeDeleted != null) {
+            System.out.println("Number of users owning the preset: " + presetToBeDeleted.getEntityUsers());
+            if (presetToBeDeleted.getEntityUsers().isEmpty() ||
+                    planItUserId == null ||
+                    (
+                            presetToBeDeleted.getEntityUsers().size() == 1 &&
+                            Objects.equals(presetToBeDeleted.getEntityUsers().stream().toList().get(0).getUser_id(), planItUserId)
+                    )
+            ) {
+                if (!presetToBeDeleted.getEntityUsers().isEmpty()) {
+                    Set<Entity_User> updatedUsers = new HashSet<>();
+                    for (Entity_User presetOwner : presetToBeDeleted.getEntityUsers()) {
+                        presetOwner.removePreset(presetToBeDeleted);
+                        updatedUsers.add(presetOwner);
+                    }
+                        idaoUser.saveAll(updatedUsers);
+                }
+                System.out.println("deleting all instances of the event preset");
+                idaoEventPreset.delete(presetToBeDeleted);
+            } else {
+                Entity_User presetOwner = idaoUser.findById(planItUserId).orElse(null);
+                if (presetOwner != null) {
+                    presetOwner.removePreset(presetToBeDeleted);
+                    idaoUser.save(presetOwner);
+                } else {
+                    //TODO:  THROW USER NOT FOUND
+                }
+            }
+        } else {
```

Comment:
I suggest escaping early to make it more readable and less nested

Reply:
The reason there is no code right now is because there will error handling in the future